pkg/auth/dependency/oauth/handler/handler_authz.go

```diff
@@ -167,7 +167,7 @@ func (h *AuthorizationHandler) validateRequest(r protocol.AuthorizationRequest)
 	if r.ResponseType() != "code" {
 		return protocol.NewError("unsupported_response_type", "only 'code' response type is supported")
 	}
-	if r.Scope() == "" {
+	if len(r.Scope()) == 1 && r.Scope()[0] == "" {
```

Comment:
`r.Scope()` should return nil or empty slice if `scope` is absent or `scope=`. The current behavior is too surprising. This applies to `ui_locales` as well.

Reply:
Fixed.